src/commands/generateLicenseKey.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2020 Make Positive Provar Ltd
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.md file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as vscode from 'vscode';
+import { messages } from '../messages';
+import { HttpService } from '../utils/httpService';
+
+const DEFAULT_VALIDATE_INPUT = (value: string) => {
+    return value === '' ? messages.generate_license_error_empty_field : null;
+};
+
+const VALIDATE_PHONE_INPUT = (value: string) => {
+    const regex = /^\d+$/;
+    return !regex.test(value) && value.length
+        ? messages.generate_license_error_incorrect_contact
+        : value.length > 16
+        ? messages.generate_license_error_contact_length
+        : null;
+};
+
+const VALIDATE_EMAIL = (value: string) => {
+    const regex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    return regex.test(value.toLowerCase()) ? null : messages.generate_license_error_incorrect_email;
+};
+
```

Comment:
Where did you get that regular expression from?

Reply:
From file:
> `/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/`
From Emailregex.com
> `/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/`

Reply:
From file:
`/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/`

Emailregex.com
`/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/`

The only difference is that the one in file escapes left side square brackets.

Incidentally I think the W3C one below it is slightly more elegant.

`/^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/`